BLSQ/iaso
# IA-3793 validate xlsform question names uniqueness

A lot of features assume uniqueness of question names

Related JIRA tickets : IA-3793

TODO discuss how to display the validation errors ?

## Self proofreading checklist


## Doc

Tell us where the doc can be found (docs folder, wiki, in the code...).

## Changes

Drop support for xls in favor of xlsx (note xls is no more supported by other platform like ona)

Add validations on 
- question name 
  - uniqueness (not if there are some repeat_group)
  - not containing blank
- using in formula known question name (relevant, calculation,...)
- referencing in select_one, select_multiple know list_name in choices

## How to test

launch the seed command 

upload forms from the jira or the fixtures folder see iaso/tests/fixtures/odk_invalid_xlsform.xlsx

## Print screen / video

Upload here print screens or videos showing the changes.

## Notes

we don't want to support xls only xlsx which is the norm (and not no more working on ona also)

## Follow the Conventional Commits specification

The **merge message** of a pull request must follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.

This convention helps to automatically generate release notes.

Use lowercase for consistency.

[Example](https://github.com/BLSQ/iaso/commit/8b8d7d3064138c1e57878f17b4eb922516ab0112):

```
fix: validate xlsform when creating form version

Refs: IA-3793
```

Note that the Jira reference is preceded by a _line break_.

Both the line break and the Jira reference are entered in the _Add an optional extended description…_ field.
